tests/unit/python/foglamp/plugins/south/coap_listen/test_coap_listen.py

```diff
@@ -0,0 +1,224 @@
+import copy
+import json
+import pytest
+import asyncio
+from unittest import mock
+from unittest.mock import call, patch
+from aiohttp.test_utils import make_mocked_request
+from aiohttp.streams import StreamReader
+from multidict import CIMultiDict
+import cbor2
+from aiocoap import message, numbers
+
+from foglamp.plugins.south.coap_listen import coap_listen
+from foglamp.plugins.south.coap_listen.coap_listen import CoAPIngest
+from foglamp.plugins.south.coap_listen.coap_listen import Ingest
+
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+def _DEFAULT_CONFIG(unused_port):
+    return {
+            'plugin': {
+                 'description': 'Python module name of the plugin to load',
+                 'type': 'string',
+                 'default': 'coap_listen'
+            },
+            'port': {
+                'description': 'Port to listen on',
+                'type': 'integer',
+                'default': str(unused_port),
+            },
+            'uri': {
+                'description': 'URI to accept data on',
+                'type': 'string',
+                'default': 'sensor-values',
+            }
+    }
+
+_NEW_CONFIG = {
+    'plugin': {
+         'description': 'Python module name of the plugin to load',
+         'type': 'string',
+         'default': 'coap_listen'
+    },
+    'port': {
+        'description': 'Port to listen on',
+        'type': 'integer',
+        'default': '1234',
+    },
+    'uri': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': 'sensor-values',
+    }
+}
+
+
+def test_plugin_info():
+    assert coap_listen.plugin_info() == {
+            'name': 'CoAP Server',
+            'version': '1.0',
+            'mode': 'async',
+            'type': 'south',
+            'interface': '1.0',
+            'config': _DEFAULT_CONFIG(5683)
+    }
+
+def test_plugin_init(unused_port):
+    config = _DEFAULT_CONFIG(unused_port())
+    assert coap_listen.plugin_init(config) == config
+
+def test_plugin_start(mocker, unused_port):
+    # GIVEN
+    config = copy.deepcopy(_DEFAULT_CONFIG(unused_port()))
+    config['port']['value'] = config['port']['default']
+    config['uri']['value'] = config['uri']['default']
+
+    log_exception = mocker.patch.object(coap_listen._LOGGER, "exception")
+    log_info = mocker.patch.object(coap_listen._LOGGER, "info")
+    assert coap_listen.aiocoap_ctx is None
+
+    # WHEN
+    loop = asyncio.get_event_loop()
+    loop.call_soon(coap_listen.plugin_start(config))
+
+    # THEN
+    assert coap_listen.aiocoap_ctx is not None
+    assert 1 == log_info.call_count
+    calls = [call('CoAP listener started on port {} with uri {}'.format(config['port']['value'], config['uri']['value']))]
+    log_info.assert_has_calls(calls, any_order=True)
+
+def test_plugin_reconfigure(mocker, unused_port):
```

Comment:
This tests a single path: when the config is different. Shouldn't the test also test the second path: when the config is the same?

Reply:
Not sure what you meant here. The reconfigure will simply stop the existing plugin instance and will restart the new plugin instance irrespective of whether the old config and new config are same or different.